packages/plugins/src/foundry/utils/index.ts

```diff
@@ -382,24 +382,34 @@ export const makeGetConfigArtifacts = (
     const toReadFiles: string[] = []
     const localBuildInfoCache = {}
 
-    const fullyQualifiedNamePromises = fullyQualifiedNames.map(
-      async (fullyQualifiedName) => {
-        const artifact = await readContractArtifact(
-          fullyQualifiedName,
-          projectRoot,
-          artifactFolder
-        )
-
+    const fullyQualifiedNamePromises = initCodeWithArgsArray.map(
```

Comment:
This section works the same as before, except:
1. We now iterate through the input `initCodeWithArgsArray` instead of the input fully qualified names, and
2. We search for a matching artifact using `isInitCodeMatch` instead of searching for the most recent build info file that includes the fully qualified name.